### PR TITLE
[ENH] Interface to TinyTimeMixer foundation model

### PR DESCRIPTION
**Pull request by geetu040**
Closes https://github.com/sktime/sktime/issues/6698, adds TinyTimeMixer as an `sktime` model.

The estimator can be used with vendored package, or a source installation.

### CODE REVIEW

**Comment by geetu040** [comment]
after having tested locally, I now want to run tests on github ci, but I cannot do that because of the dependency.
`tsfm_public` is not available on pypi and therefore is installed like this
```cmd
pip install git+https://github.com/ibm-granite/granite-tsfm.git@v0.2.3
```
@benHeid @fkiraly How would you suggest we cater this problem?

File: docs/source/api_reference/forecasting.rst [hunk 1]
```diff
@@ -439,6 +439,14 @@ Deep learning based forecasters
 
     HFTransformersForecaster
 
+.. currentmodule:: sktime.forecasting.ttm
```
Comment (fkiraly) [inline docs/source/api_reference/forecasting.rst:442]: introduce a separate category maybe, or split?

There are enough DL forecasters to split them, high-level we have 1st generation recurrence architectures, 2nd generation transformer architectures, and 3rd generation zero-shot/few-shot transformers, aka foundation models.

Not a blocker, but I think for user experience might be worth considering. FYI @benHeid, how would you split this?

File: sktime/forecasting/tests/test_all_forecasters.py [hunk 2]
```diff
@@ -931,7 +931,16 @@ def test_pridect_signature(self, estimator_class):
         assert "X" in sig.parameters.keys()
         assert "y" in sig.parameters.keys()
 
+    def _check_global_tag(self, estimator_instance):
```
Comment (fkiraly) [inline sktime/forecasting/tests/test_all_forecasters.py:934]: where is this coming from, why are we changing the tests?

Best practice is to not modify the interface tests while adding an object of an interface.

File: sktime/libs/granite_ttm/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (fkiraly) [inline sktime/libs/granite_ttm/LICENSE:1]: redundant with the `__init__`?

File: sktime/libs/granite_ttm/__init__.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
+"""Python implementation TinyTimeMixer Foundation Model.
+
+Unofficial fork of the ``ibm-granite/granite-tsfm`` package, maintained in ``sktime``.
```
Comment (fkiraly) [inline sktime/libs/granite_ttm/__init__.py:3]: perhaps add, temporary with a view to moving to a potential package release of ttm

File: sktime/forecasting/ttm.py [hunk 5]
```diff
@@ -0,0 +1,569 @@
...
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "capability:pred_int": False,
+        "capability:pred_int:insample": False,
+        "authors": ["geetu040"],
```
Comment (fkiraly) [inline sktime/forecasting/ttm.py:136]: add ttm authors

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great, thanks!

Left some comments related to documentation mostly.

File: sktime/forecasting/ttm.py (hunk 5 above)
Comment (geetu040) [reply sktime/forecasting/ttm.py]: Added

File: sktime/libs/granite_ttm/__init__.py (hunk 4 above)
Comment (geetu040) [reply sktime/libs/granite_ttm/__init__.py:3]: Added

File: sktime/libs/granite_ttm/LICENSE (hunk 3 above)
Comment (geetu040) [reply sktime/libs/granite_ttm/LICENSE]: yes, its redundant, I've removed the LICENSE file

File: sktime/forecasting/tests/test_all_forecasters.py (hunk 2 above)
Comment (geetu040) [reply sktime/forecasting/tests/test_all_forecasters.py:934]: Source: https://github.com/sktime/sktime/pull/6686
These editions are made in all global forecasting PRs, as it is important and avoids certain test cases that should not be performed on global estimators.

File: docs/source/api_reference/forecasting.rst (hunk 1 above)
Comment (geetu040) [reply docs/source/api_reference/forecasting.rst:442]: Just sketching some ideas
- MLP
	- LTSFLinearForecaster
	- LTSFDLinearForecaster
	- LTSFNLinearForecaster

- Recurrent
	- NeuralForecastRNN
	- NeuralForecastLSTM

- Transformer / Pre-trained
	- HFTransformersForecaster
	- TinyTimeMixerForecaster
	- LTSFTransformerForecaster

- Foundation / Pre-trained
	- TimesFM

- not sure about how they work
	- PyKANForecaster
	- PytorchForecastingTFT
	- PytorchForecastingNBeats
	- CINNForecaster

File: sktime/forecasting/tests/test_all_forecasters.py (hunk 2 above)
Comment (Xinyu-Wu-0000) [reply sktime/forecasting/tests/test_all_forecasters.py:934]: It's becasue the `broadcasting` parameter. If `broadcasting==True`, the global forecasting ability will be turned off in `__init__`. So some of the `estimator_instance`s do not have the global forecasting ability.

**Review by fkiraly — DISMISSED** [review]
Great!

**Review by fkiraly — APPROVED** [review]
Thanks!

**Pull request merged**